src/wp-includes/class-wp-theme-json.php

```diff
@@ -3673,6 +3686,35 @@ protected static function remove_insecure_inner_block_styles( $blocks ) {
 		return $sanitized;
 	}
 
+	/**
+	 * Preserves valid typed settings from input to output based on type markers in schema.
+	 *
+	 * Recursively iterates through the schema and validates/preserves settings
+	 * that have type markers (e.g., boolean) in VALID_SETTINGS.
+	 *
+	 * @since 7.0.0
+	 *
+	 * @param array $input  Input settings to process.
+	 * @param array $output Output settings array (passed by reference).
+	 * @param array $schema Schema to validate against (typically VALID_SETTINGS).
+	 * @param array $path   Current path in the schema (for recursive calls).
```

Comment:
This seems like an opportunity to supply a more specific type for the `$path` parameter, like `array<string|int>`:

```suggestion
	 * @param array             $input  Input settings to process.
	 * @param array             $output Output settings array (passed by reference).
	 * @param array             $schema Schema to validate against (typically VALID_SETTINGS).
	 * @param array<string|int> $path   Current path in the schema (for recursive calls).
```

At first I was thinking `string[]`, but in looking at `_wp_array_set()`, it has this logic:

https://github.com/WordPress/wordpress-develop/blob/a1071476c73e110d214b1c25cba0b27ebb506386/src/wp-includes/functions.php#L5171-L5178

Curiously, there is no `_doing_it_wrong()` here. Also curious is that that `null` is apparently allowed among the `$path` array? This seems wrong.

Reply:
Good spotting. yeah it seems weird that `$path = array('a', null, 'b')` is allowed.

Looks like the tests don't cover it either

https://github.com/WordPress/wordpress-develop/blob/a1071476c73e110d214b1c25cba0b27ebb506386/tests/phpunit/tests/functions/wpArraySet.php